base/abstractarray.jl

```diff
@@ -488,7 +527,7 @@ function circshift{T,N}(a::AbstractArray{T,N}, shiftamts)
     for i=1:N
         s = size(a,i)
         d = i<=length(shiftamts) ? shiftamts[i] : 0
-        I = tuple(I..., d==0 ? [1:s] : mod([-d:s-1-d], s).+1)
+        I = tuple(I..., d==0 ? [1:s;] : mod([-d:s-1-d;], s).+1)
```

Comment:
Just to check: in this PR, is `[1:s;]` equivalent to `[1:s...]` (or maybe `[(1:s)...]`, if the attachment of the elipses is ambiguous)?